### PR TITLE
Feature/clientside error recovery

### PR DESCRIPTION
**Pull request by aCinal**

### CODE REVIEW

File: client/tests/test_malformed_messages.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    # Instantiate the session manager
+    sm = SessionManager(generate_keys(), Account())
+
+    # Create a malformed message
+    message = CansMessage()
+    message.header.sender = digest_key(generate_keys()[1])
+    message.header.receiver = sm.identity
+    message.header.msg_id = CansMsgId.PEER_LOGIN
+    message.payload = {
+        "unexpected": "For nothing can seem foul to those that win"
```
Comment (GabrielWechta) [inline client/tests/test_malformed_messages.py:24]: Then with the losers let it sympathize. As now we meet.

File: common/src/common/messages.py [hunk 2]
```diff
@@ -106,14 +107,50 @@ def __init__(self, receiver: str) -> None:
...
+
 class SchnorrCommit(CansMessage):
     """Commitment of the Schnorr identification scheme."""
 
     def __init__(self, public_key: str, commitment: str) -> None:
         """Create a Schnorr commitment message."""
         super().__init__()
         self.header.msg_id = CansMsgId.SCHNORR_COMMIT
-        self.header.receiver = ""
+        self.header.receiver = None
```
Comment (GabrielWechta) [inline common/src/common/messages.py:153]: receiver by default is None, right? Or do you want to emphasize that? :)

File: integration/test_add_friends.py [hunk 3]
```diff
@@ -55,13 +55,13 @@ async def add_friend_and_wait_for_login_notification(
     ) -> None:
         """Add friend and wait for PEER_LOGIN message."""
         self.log.debug(
-            f"{self.session_manager.identity} sending ADD_FRIEND"
-            + "message to the server..."
+            f"'{self.session_manager.identity}' sending ADD_FRIEND"
```
Comment (GabrielWechta) [inline integration/test_add_friends.py:58]: I really like those " ' ", a shame we didn't add it before

File: integration/test_session_error.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        name="Bob",
+        keys=(bob_secret, bob_public),
+        friends={digest_key(alice_public), digest_key(mallory_public)},
+        crypto_session_peer_override=None,
+    )
+    mallory = MockClient(
+        name="Charlie",
+        keys=(mallory_secret, mallory_public),
+        friends={digest_key(alice_public), digest_key(bob_public)},
+        # Let Mallory be the broken client that always uses his session
```
Comment (GabrielWechta) [inline integration/test_session_error.py:146]: this one https://pl.wikipedia.org/wiki/George_Mallory ? :)
I guess in the high mountains one may get confused and change his name from George to Mallory

File: run_full_test_suite [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (GabrielWechta) [inline run_full_test_suite:1]: Don't we want to use some virtual env for this?

**Review by GabrielWechta — APPROVED** [review]
Really nice commit, thank you for improving my docstrings, much better.
Also, test for session error - great!

File: common/src/common/messages.py (hunk 2 above)
Comment (aCinal) [reply common/src/common/messages.py:153]: Just to emphasize :)

File: client/tests/test_malformed_messages.py (hunk 1 above)
Comment (aCinal) [reply client/tests/test_malformed_messages.py:24]: <3

File: run_full_test_suite (hunk 5 above)
Comment (aCinal) [reply run_full_test_suite:1]: Quick and dirty script for the lazy :) to be improved later

File: integration/test_session_error.py (hunk 4 above)
Comment (aCinal) [reply integration/test_session_error.py:141]: <3

**Review by GabrielWechta — APPROVED** [review]
👍

**Pull request merged**